File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int maxdiff=0;
        int minval = prices[0];
        for(int i=0;i<prices.size();i++){
            maxdiff= max(maxdiff,prices[i]-minval);
            minval = min(prices[i],minval);
        }
        return maxdiff;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        maxdiff = 0
        minval = prices[0]
        for i in range(1,len(prices)):
            maxdiff = max(maxdiff,prices[i]-minval)
            minval = min(minval,prices[i])
        return maxdiff